File: frontend/src/services/interviews.js
const API_BASE_URL = import.meta.env.VITE_API_BASE_URL || '/api';

class InterviewService {
  getToken() {
    return sessionStorage.getItem('medikiosk_access_token');
  }
  
  async createInterview(consultationId) {
    const token = this.getToken();
    if (!token) return { success: false, error: 'Not authenticated' };
    
    try {
      const response = await fetch(`${API_BASE_URL}/interviews`, {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ consultation_id: consultationId }),
      });
      
      const data = await response.json();
      
      if (!response.ok) {
        return { success: false, error: data.detail || 'Failed to create interview' };
      }
      
      return { success: true, data: data.data };
    } catch (error) {
      return { success: false, error: 'Network error. Please try again.' };
    }
  }
  
  async getInterview(interviewId) {
    const token = this.getToken();
    if (!token) return { success: false, error: 'Not authenticated' };
    
    try {
      const response = await fetch(`${API_BASE_URL}/interviews/${interviewId}`, {
        headers: {
          'Authorization': `Bearer ${token}`,
        },
      });
      
      const data = await response.json();
      
      if (!response.ok) {
        return { success: false, error: data.detail || 'Failed to load interview' };
      }
      
      return { success: true, data: data.data };
    } catch (error) {
      return { success: false, error: 'Network error. Please try again.' };
    }
  }
  
  async startInterview(interviewId) {
    const token = this.getToken();
    if (!token) return { success: false, error: 'Not authenticated' };
    
    try {
      const response = await fetch(`${API_BASE_URL}/interviews/${interviewId}/start`, {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
        },
      });
      
      const data = await response.json();
      
      if (!response.ok) {
        return { success: false, error: data.detail || 'Failed to start interview' };
      }
      
      return { success: true };
    } catch (error) {
      return { success: false, error: 'Network error. Please try again.' };
    }
  }
  
  async submitResponse(interviewId, questionId, responseText, inputMethod = 'text') {
    const token = this.getToken();
    if (!token) return { success: false, error: 'Not authenticated' };
    
    try {
      const response = await fetch(`${API_BASE_URL}/interviews/${interviewId}/responses`, {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          question_id: questionId,
          response_text: responseText,
          input_method: inputMethod,
        }),
      });
      
      const data = await response.json();
      
      if (!response.ok) {
        return { success: false, error: data.detail || 'Failed to submit response' };
      }
      
      return { success: true, data: data.data };
    } catch (error) {
      return { success: false, error: 'Network error. Please try again.' };
    }
  }
  
  async pauseInterview(interviewId) {
    const token = this.getToken();
    if (!token) return { success: false, error: 'Not authenticated' };
    
    try {
      const response = await fetch(`${API_BASE_URL}/interviews/${interviewId}/pause`, {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
        },
      });
      
      const data = await response.json();
      
      if (!response.ok) {
        return { success: false, error: data.detail || 'Failed to pause interview' };
      }
      
      return { success: true };
    } catch (error) {
      return { success: false, error: 'Network error. Please try again.' };
    }
  }
  
  async cancelInterview(interviewId) {
    const token = this.getToken();
    if (!token) return { success: false, error: 'Not authenticated' };
    
    try {
      const response = await fetch(`${API_BASE_URL}/interviews/${interviewId}/cancel`, {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
        },
      });
      
      const data = await response.json();
      
      if (!response.ok) {
        return { success: false, error: data.detail || 'Failed to cancel interview' };
      }
      
      return { success: true };
    } catch (error) {
      return { success: false, error: 'Network error. Please try again.' };
    }
  }
}

export const interviewService = new InterviewService();

File: backend/app/api/endpoints/interviews.py
"""
Interview API endpoints.
"""
from fastapi import APIRouter, Depends, HTTPException, status
from app.api.dependencies.auth import require_authenticated_user
from app.models.interview import InterviewCreate, InterviewResponse
from app.services.interview_service import interview_service
import logging

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/interviews", tags=["interviews"])

@router.post("")
async def create_interview(
    interview_data: InterviewCreate,
    current_user: dict = Depends(require_authenticated_user),
):
    """Create interview session."""
    result = await interview_service.create_interview(
        current_user,
        interview_data.consultation_id,
    )
    
    if not result["success"]:
        raise HTTPException(
            status_code=result["status_code"],
            detail=result["error"],
        )
    
    interview = result["interview"]
    return {
        "success": True,
        "message": result["message"],
        "data": {
            "interview_id": str(interview["_id"]),
            "consultation_id": interview["consultation_id"],
            "status": interview["status"],
        }
    }

@router.get("/{interview_id}")
async def get_interview(
    interview_id: str,
    current_user: dict = Depends(require_authenticated_user),
):
    """Get interview state."""
    result = await interview_service.get_interview(current_user, interview_id)
    
    if not result["success"]:
        raise HTTPException(
            status_code=result["status_code"],
            detail=result["error"],
        )
    
    return {
        "success": True,
        "data": {
            "interview": result["interview"],
            "responses": result["responses"],
            "current_question": result["current_question"],
            "progress": result["progress"],
        }
    }

@router.post("/{interview_id}/start")
async def start_interview(
    interview_id: str,
    current_user: dict = Depends(require_authenticated_user),
):
    """Start interview."""
    result = await interview_service.start_interview(current_user, interview_id)
    
    if not result["success"]:
        raise HTTPException(
            status_code=result["status_code"],
            detail=result["error"],
        )
    
    return {"success": True, "message": result["message"]}

@router.post("/{interview_id}/responses")
async def submit_response(
    interview_id: str,
    response_data: InterviewResponse,
    current_user: dict = Depends(require_authenticated_user),
):
    """Submit interview response."""
    result = await interview_service.submit_response(
        current_user,
        interview_id,
        response_data.question_id,
        response_data.response_text,
        response_data.input_method,
    )
    
    if not result["success"]:
        raise HTTPException(
            status_code=result["status_code"],
            detail=result["error"],
        )
    
    return {
        "success": True,
        "message": result["message"],
        "data": result,
    }

@router.post("/{interview_id}/pause")
async def pause_interview(
    interview_id: str,
    current_user: dict = Depends(require_authenticated_user),
):
    """Pause interview."""
    result = await interview_service.pause_interview(current_user, interview_id)
    
    if not result["success"]:
        raise HTTPException(
            status_code=result["status_code"],
            detail=result["error"],
        )
    
    return {"success": True, "message": result["message"]}

@router.post("/{interview_id}/resume")
async def resume_interview(
    interview_id: str,
    current_user: dict = Depends(require_authenticated_user),
):
    """Resume interview."""
    result = await interview_service.resume_interview(current_user, interview_id)
    
    if not result["success"]:
        raise HTTPException(
            status_code=result["status_code"],
            detail=result["error"],
        )
    
    return {"success": True, "message": result["message"]}

@router.post("/{interview_id}/cancel")
async def cancel_interview(
    interview_id: str,
    current_user: dict = Depends(require_authenticated_user),
):
    """Cancel interview."""
    result = await interview_service.cancel_interview(current_user, interview_id)
    
    if not result["success"]:
        raise HTTPException(
            status_code=result["status_code"],
            detail=result["error"],
        )
    
    return {"success": True, "message": result["message"]}